src/BlockEntities/DispenserEntity.cpp

```diff
@@ -178,6 +189,64 @@ void cDispenserEntity::DropSpenseFromSlot(cChunk & a_Chunk, int a_SlotNum)
 			break;
 		}
 
+		case E_ITEM_BOTTLE_O_ENCHANTING:
+		{
+			if (SpawnProjectileFromDispenser(BlockX, DispY, BlockZ, cProjectileEntity::pkExpBottle, GetShootVector(Meta) * 20 + Vector3d(0, 1, 0)) != cEntity::INVALID_ID)
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
+		}
+
+		case E_ITEM_POTION:
+		{
+			if (SpawnProjectileFromDispenser(BlockX, DispY, BlockZ, cProjectileEntity::pkSplashPotion, GetShootVector(Meta) * 20 + Vector3d(0, 1, 0), &SlotItem) != cEntity::INVALID_ID)
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
+		}
+
+		case E_ITEM_DYE:
+		{
+			if (SlotItem.m_ItemDamage != E_META_DYE_WHITE)
+			{
+				DropFromSlot(a_Chunk, a_SlotNum);
+				break;
+			}
+			if (m_World->GrowRipePlant(BlockX, DispY, BlockZ, true))
+			{
+				m_Contents.ChangeSlotCount(a_SlotNum, -1);
+			}
+			break;
```

Comment:
Bonemeal does indeed never drop in Vanilla from a dispenser.

Bonemeal is one of the many dyes. The bonemeal functions should only be used, if the item is actually bonemeal. If the item metadata is not "E_META_DYE_WHITE", than it is another dye which is not bonemeal.